File: JavaScript/2024/Jan/Q1207-UniqueNumberOfOccurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function (arr) {
    const occur = {};
    for (let i = 0; i < arr.length; i++) {
        occur[arr[i]] = (occur[arr[i]] || 0) + 1;
    }

    return new Set(Object.values(occur)).size === new Set(arr).size;
};

File: Python/Leetcode75/Easy/Hashmap/Q1207-UniqueNumberOfOccurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        m = {}
        fq = set()

        for num in arr:
            m[num] = m.get(num, 0) + 1
        
        for _, f in m.items():
            fq.add(f)
        
        return len(set(arr)) == len(fq)

